File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/RxFactoryModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import dagger.Module;
import dagger.Provides;
import com.benoitquenaudon.tvfoot.red.app.domain.match.state.MatchIntent;
import com.benoitquenaudon.tvfoot.red.app.domain.match.state.MatchViewState;
import com.benoitquenaudon.tvfoot.red.app.domain.matches.state.MatchesIntent;
import com.benoitquenaudon.tvfoot.red.app.domain.matches.state.MatchesViewState;
import io.reactivex.subjects.PublishSubject;

@Module public class RxFactoryModule {
  @Provides PublishSubject<MatchesIntent> provideMatchesIntentSubject() {
    return PublishSubject.create();
  }

  @Provides PublishSubject<MatchesViewState> provideMatchesViewStateSubject() {
    return PublishSubject.create();
  }

  @Provides PublishSubject<MatchIntent> provideMatchIntentSubject() {
    return PublishSubject.create();
  }

  @Provides PublishSubject<MatchViewState> provideMatchViewStateSubject() {
    return PublishSubject.create();
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/RxFactoryModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import com.benoitquenaudon.tvfoot.red.app.domain.match.MatchIntent
import com.benoitquenaudon.tvfoot.red.app.domain.match.MatchViewState
import com.benoitquenaudon.tvfoot.red.app.domain.matches.MatchesIntent
import com.benoitquenaudon.tvfoot.red.app.domain.matches.MatchesViewState
import dagger.Module
import dagger.Provides
import io.reactivex.subjects.PublishSubject

@Module object RxFactoryModule {
  @JvmStatic @Provides fun provideMatchesIntentSubject(): PublishSubject<MatchesIntent> {
    return PublishSubject.create<MatchesIntent>()
  }

  @JvmStatic @Provides fun provideMatchesViewStateSubject(): PublishSubject<MatchesViewState> {
    return PublishSubject.create<MatchesViewState>()
  }

  @JvmStatic @Provides fun provideMatchIntentSubject(): PublishSubject<MatchIntent> {
    return PublishSubject.create<MatchIntent>()
  }

  @JvmStatic @Provides fun provideMatchViewStateSubject(): PublishSubject<MatchViewState> {
    return PublishSubject.create<MatchViewState>()
  }
}
